app/src/main/java/org/mozilla/focus/urlinput/UrlInputFragment.kt

```diff
@@ -0,0 +1,307 @@
+        when (view.id) {
+            R.id.suggestion_item -> {
+                setUrlText((view as TextView).text)
+                TelemetryWrapper.searchSuggestionLongClick()
+                return true
+            }
+            R.id.clear -> {
+                TelemetryWrapper.searchClear()
+                return false
+            }
+            R.id.dismiss -> {
+                TelemetryWrapper.searchDismiss()
+                return false
+            }
+            else -> return false
+        }
+    }
+
+    override fun onClick(view: View) {
+        when (view.id) {
+            R.id.clear -> {
+                urlView.setText("")
+                urlView.requestFocus()
+            }
+            R.id.dismiss -> dismiss()
+            R.id.suggestion_item -> onSuggestionClicked((view as TextView).text)
+            else -> throw IllegalStateException("Unhandled view in onClick()")
+        }
+    }
+
+    private fun initByArguments() {
+        val args = arguments
+        if (args?.containsKey(ARGUMENT_URL) == true) {
+            val url = args.getString(ARGUMENT_URL)
+            urlView.setText(url)
+            clearView.visibility = if (TextUtils.isEmpty(url)) View.GONE else View.VISIBLE
+        }
+        allowSuggestion = args?.getBoolean(ARGUMENT_ALLOW_SUGGESTION, true) ?: false
+    }
+
+    private fun onSuggestionClicked(tag: CharSequence) {
+        search(tag.toString())
+    }
+
+    private fun dismiss() {
+        // This method is called from animation callbacks. In the short time frame between the animation
+        // starting and ending the activity can be paused. In this case this code can throw an
+        // IllegalStateException because we already saved the state (of the activity / fragment) before
+        // this transaction is committed. To avoid this we commit while allowing a state loss here.
+        // We do not save any state in this fragment (It's getting destroyed) so this should not be a problem.
+        val activity = activity
+        if (activity is FragmentListener) {
+            (activity as FragmentListener).onNotified(this, FragmentListener.TYPE.DISMISS_URL_INPUT, true)
+        }
+    }
+
+    private fun onCommit() {
+        val input = urlView.autocompleteResult.formattedText.let {
+            if (it.isEmpty()) urlView.text.toString() else it
+        }
+        search(input)
+    }
+
+    private fun search(input: String){
+        if (!input.trim { it <= ' ' }.isEmpty()) {
+            ViewUtils.hideKeyboard(urlView)
+
+            val isUrl = SupportUtils.isUrl(input)
+
+            val url = if (isUrl)
+                SupportUtils.normalize(input)
+            else
+                SearchUtils.createSearchUrl(context, input)
+
+            val isOpenInNewTab = openUrl(url)
+
+            if (isOpenInNewTab) {
+                TelemetryWrapper.addNewTabFromHome()
+            }
+            TelemetryWrapper.urlBarEvent(isUrl, !urlView.autocompleteResult.isEmpty)
```

Comment:
It doesn't seem to me that urlView.autocompleteResult.isEmpty is a nice proxy to isSuggestion.